### PR TITLE
Add error handling from v 12

### PR DESCRIPTION
**Pull request by ericschlanger**

### CODE REVIEW

File: lib/synapse_pay_rest/http_client.rb [hunk 1]
```diff
@@ -52,49 +52,75 @@ def update_headers(user_id: nil, oauth_key: nil, fingerprint: nil, client_id: ni
...
+      # are the HTTP Client responses.
+      case e.response.code
+      when 400
+        return e.response
+      when 401
+        return e.response
+      when 409
+        return e.response
+      when 500
+        return e.response
```
Comment (dlm) [inline lib/synapse_pay_rest/http_client.rb]: I think there may be a bug in the v12 code.  When we were looking over the errors, did we find that the 500 errors were returning html?  Maybe it returns both?

File: lib/synapse_pay_rest/http_client.rb [hunk 2]
```diff
@@ -52,49 +52,75 @@ def update_headers(user_id: nil, oauth_key: nil, fingerprint: nil, client_id: ni
...
       end
-      return JSON.parse(response)
+    end
+
+    def handle_method_not_allowed()
+      return {'success' => false, 'reason' => 'The method is not allowed. Check your id parameters.'}.to_json
+    end
+
+    def handle_gateway_error()
+      return {'success' => false, 'reason' => 'The gateway appears to be down.  Check synapsepay.com or try again later.'}.to_json
```
Comment (dlm) [inline lib/synapse_pay_rest/http_client.rb]: We may want to modify the error messages here (depending on how they will present to the user)

File: lib/synapse_pay_rest/http_client.rb [hunk 3]
```diff
@@ -52,49 +52,75 @@ def update_headers(user_id: nil, oauth_key: nil, fingerprint: nil, client_id: ni
...
+    end
+
+    def handle_timeout_error()
+      return {'success' => false, 'reason' => 'A timeout has occurred.'}.to_json
+    end
+
+    def handle_unknown_error()
+      return {'success' => false, 'reason' => 'Unknown error in library. Contact synapsepay.'}.to_json
```
Comment (dlm) [inline lib/synapse_pay_rest/http_client.rb]: Similar to above

**Comment by dlm** [comment]
Looks like a big improvement from the error handling before.  I added a few comments, but when ready :shipit:

**Comment by ericschlanger** [comment]
@dlm I updated the error messages and also fixed the bug with assuming 500 will be JSON

**Pull request merged**